cfspopcon/unit_handling/default_units.py

```diff
@@ -18,11 +17,10 @@
 def check_units_are_valid(units_dictionary: dict[str, str]) -> None:
     """Ensure that all units in units_dictionary are valid."""
     invalid_units = []
-    for key, units in units_dictionary.items():
+    for units in units_dictionary.values():
         try:
             Quantity(1.0, units)
         except UndefinedUnitError:  # noqa: PERF203
-            warn(f"Undefined units '{units}' for '{key}", stacklevel=3)
```

Comment:
Can you still tell which key has invalid units? This is meant to summarise all invalid entries. If you just want to throw an error on the first one, can get rid of the try/except.

Reply:
@tbody-cfs I've pushed a commit changing the thrown error to include the `variable_name: unit` pairs that raised an error. 

Let me know what you think